### ISSUE TITLE
-no-console option does not work when Windows Terminal is set as Default Terminal application

### ISSUE THREAD

**Issue by dasbootbee** [syncthing/syncthing#8046]
### Does your log mention database corruption?

No

### Include required information

v1.18.4, Windows 11

Running Windows Terminal 1.11.2921.0
Under settings, set Default Terminal application to "Windows Terminal"
Run syncthing using the following:
syncthing.exe -no-console -no-browser
Console window opens

Set Default Terminal application to "Windows Console Host"
Works as expected (no console window appears)

**Comment by AudriusButkevicius** [comment]
I don't think we can do anything here.

Windows terminal is not a real console, as far as I understand it's a simple application that emulates to be a console, yet doesn't seem to implement the hide feature etc.

I think you should raise it with the people maintaining the terminal, yet I am not sure how much they will be able to do.

**Comment by bt90** [comment]
I think we might get an influx of people complaining about this in the near future: https://blogs.windows.com/windows-insider/2022/08/24/announcing-windows-11-insider-preview-build-25188/

tl;dr Windows Terminal is going to be the default with the upcoming Windows 11 update

**Comment by calmh** [comment]
My impression is that this is being fixed in the terminal, if this means what I think it means

https://github.com/microsoft/terminal/pull/12515

**Comment by whalehub** [comment]
> I think we might get an influx of people complaining about this in the near future: https://blogs.windows.com/windows-insider/2022/08/24/announcing-windows-11-insider-preview-build-25188/
> 
> tl;dr Windows Terminal is going to be the default with the upcoming Windows 11 update

This issue popped up for me after updating Windows 11 (non-preview) today. The workaround mentioned by @BennD works fine for the time being.

**Comment by bt90** [comment]
@whalehub could you check your version of the terminal app? The current [release](https://github.com/microsoft/terminal/releases/tag/v1.15.2874.0) (2022-10-18) should include https://github.com/microsoft/terminal/pull/12515

**Comment by whalehub** [comment]
> @whalehub could you check your version of the terminal app? The current [release](https://github.com/microsoft/terminal/releases/tag/v1.15.2874.0) (2022-10-18) should include [microsoft/terminal#12515](https://github.com/microsoft/terminal/pull/12515)

That's the version I'm using right now.

**Comment by DonBureq** [comment]
i managed to fix that by using conhost instead of cmd.exe which redirects to tw.exe when you have Windows terminal selected as default console.

**Comment by randomvideosyoutube** [comment]
I managed to find a pretty simple fix to this after researching a bit, I had issues running other programs due to the legacy version of cmd so decided to search for another way to make it work on the normal version of cmd.

1) Make a shortcut of the "syncthing.exe" file

2) Go to the properties of the shortcut and add "--no-browser" after the directory in the 'Target' 

![shortcut](https://i.imgur.com/HW0AFNF.png)

3) Open notepad and paste the following script 

Set WshShell = CreateObject("WScript.Shell") 
WshShell.Run chr(34) & "[PASTE DIRECTORY OF SHORTCUT HERE]" & Chr(34), 0
Set WshShell = Nothing

4) Make sure the directory of the shortcut has the ".lnk" extension, if not add it.

![Notepad lnk](https://i.imgur.com/9DoBGDF.png)

5) Save the file as ".vbs" and run that

Now you should be able to use the normal cmd terminal rather than switching to the legacy.

**Comment by arthniwa** [comment]
For the `conhost` solution, more specifically, the target field of the shortcut could be something like `C:\Windows\System32\conhost.exe "C:\Program Files\Syncthing\syncthing.exe" -no-console -no-browser`.

**Comment by ionreflex** [comment]
I feel like I need to share this: I've been using Windows Terminal for a long time, never had this issue with the console showing before v1.23.6; v1.23.5 is still present on my deck and the console won't show when I use it, no need for the "Use Legacy Console" option (which works with v1.23.6, yeah...) I run Windooze 10 Entreprise

**Comment by dgasaway** [comment]
I'll add that the solution given by @arthniwa also works for Task Scheduler.

**Comment by M1saka10010** [comment]
> In my case, @BennD workaround isn't good enough because it overrides the global setting. IMO, default CMD is such a pain in everything else than disappearing in the background...
> 
> > @whalehub could you check your version of the terminal app? The current [release](https://github.com/microsoft/terminal/releases/tag/v1.15.2874.0) (2022-10-18) should include [microsoft/terminal#12515](https://github.com/microsoft/terminal/pull/12515)
> 
> Mine is 1.15.2875.0 and the issue is still present.
> 
> Switching to task scheduler doesn't do the trick either (for me at least) since it appears to be impossible for a user (even in administrators group) to schedule a "run whether user is logged on or not" task. I don't know if it's a locale related conflict within windows but I've been (as many others according to microsoft forums) experiencing this issue for years (win 10 and now win 11). Only option available seems to be "Run only when user is logged on" and the terminal window is still popping-up.
> 
> For those who know a little dotnet, my workaround consists in using a very simple console app as a launcher wrapper.
> 
> ```
> // Program.cs
> 
> using System.Diagnostics;
> 
> string program = Path.GetFullPath(Environment.GetCommandLineArgs()[1]);
> var processInfo = new ProcessStartInfo
> {
>     CreateNoWindow = true,
>     WindowStyle = ProcessWindowStyle.Hidden,
>     UseShellExecute = false,
>     FileName = program,
>     WorkingDirectory = Path.GetDirectoryName(program),
>     Arguments = string.Join(" ", Environment.GetCommandLineArgs()[2..]),
> };
> try
> {
>     using var process = Process.Start(processInfo);
>     await process.WaitForExitAsync();
>     return 0;
> }
> catch (Exception ex)
> {
>     Console.WriteLine(ex.Message);
>     return -1;
> }
> ```
> 
> ```
> // LauncherWrapper.csproj
> 
> <Project Sdk="Microsoft.NET.Sdk">
>   <PropertyGroup>
>     <OutputType>WinExe</OutputType>
>     <TargetFramework>net6.0</TargetFramework>
>     <ImplicitUsings>enable</ImplicitUsings>
>   </PropertyGroup>
> </Project>
> ```
> 
> The important thing is to set `<OutputType>WinExe</OutputType>` which will allow the wrapper to run silently.
> 
> After publishing it, all you have to do is using it in the scheduler : `C:\Path\To\LauncherWrapper.exe C:\Path\To\syncthing.exe serve --no-browser --no-console` Voilà !

I have translated it into a Python version, save the following code as `start.py`:

```python
# start.py

import os
import sys
import subprocess

CREATE_NO_WINDOW = 0x08000000

program = os.path.abspath(sys.argv[1])
process_info = subprocess.Popen(
    [program, *sys.argv[2:]],
    stdout=subprocess.PIPE,
    stderr=subprocess.PIPE,
    shell=False,
    creationflags=CREATE_NO_WINDOW,
    cwd=os.path.dirname(program),
)

try:
    process_info.wait()
    return_code = process_info.returncode
    if return_code == 0:
        print("Process completed successfully.")
    else:
        print(f"Process exited with return code {return_code}.")

except Exception as ex:
    print(f"An error occurred: {ex}")
```
Then you can run as `pythonw C:\Path\To\start.py C:\Path\To\syncthing.exe -no-browser`
